src/app/layout.tsx

```diff
@@ -22,18 +22,25 @@
  */
 'use client';
 
+import { ReactNode } from 'react';
 import { Work_Sans } from 'next/font/google';
-import ThemeProvider from '@/components/ThemeProvider';
+import { getToken } from '@/global/utils/auth';
+import { isValidJwt } from '@/lib/egoJwt';
 import Header from './components/Header';
+import ThemeProvider from './components/ThemeProvider';
 
 const workSans = Work_Sans({ subsets: ['latin'] });
 
-export default function RootLayout({ children }: { children: React.ReactNode }) {
+export default function RootLayout({ children }: { children: ReactNode }) {
+	const egoJwt = getToken() || '';
+
+	const loggedIn = Boolean(egoJwt && isValidJwt(egoJwt));
+
 	return (
 		<html lang="en">
 			<body className={workSans.className}>
 				<ThemeProvider>
-					<Header />
+					<Header loggedIn={loggedIn} />
 					{children}
 				</ThemeProvider>
```

Comment:
re slack message.

Issue is two fold:
1) `layout.tsx` is not re-rendered. there is no state change. acting as it should for SPA routing. something like the AuthProvider you had previously is the way to go I think, you can make a state change and the rest should re-render. very similar to platform-ui. I'd put it directly after `ThemeProvider` - github won't allow me to suggest commit on deleted lines - anyway - ThemeProvider will likely never re-render because we're not giving it dynamic themes where as AuthProvider. that is to say , context providers will re-render all children.  

the layout special convention component is similar to next12  `_app.tsx`
These two pages are worth a read:
- https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts
- https://nextjs.org/docs/app/api-reference/file-conventions ( some of these are VERY cool like `loading.js` )

2) we're in reactstrict mode. this means in dev renders will be called twice. its a way for react to call out things that may not be safe in production. https://react.dev/reference/react/StrictMode 
in this specific case, what if you make a network request to ego, then react re-renders the request page before you can act on the response? you get this exact double response happening. So it's making sure you handle these cases. we can bring in a library for all our network requests eg. react-query , otherwise I believe an AbortController is what you want for making sure duplicate requests are not sent

These seem like new additions above the scope of feedback so it might be best to just address PR feedback, merge, and then we can work through a more tightly scoped PR.

Reply:
Alright, react-query handles the double request issue perfectly, and I've got the AuthProvider setup as you suggested and it's working better. The Login button show/hide is still not perfect but we can improve on it later.